pooPy
=====

A collection of python scripts created for tailored purposes during research.

=====
Index of scripts:
=====

RenameFasta.py: 
    Rename sequence names in fasta file so that paired end reads have different names with different suffixes: /1, or /2.

get_contig.py:
    Given a fasta file, find the sequence for the specific region of a sequence with given ID.
    Return -1 if the sequence does not exist or its sequence does not include the specified region.

SplitSam.py:
    From a sam file generated by aligning reads/contigs to many reference seqs, by default, generate one sam file for each reference seq, each includes the reads aligning with it. This could be problematic if there are too many reference sequences, and then too many files will be open at the same time. One can specify the name of the reference of interest and extract reads mapped to this reference (along with the appropriate headers).

samStat.py:
    Get statistics of each reference sequence and mapping information of each read, from .sam file output from aligner (mapping progam).
